jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java

```diff
@@ -1647,6 +1638,22 @@ public void setServletHandler(ServletHandler servletHandler)
         relinkHandlers();
     }
 
+    @Override
+    public void setHandler(Handler handler)
+    {
+        if (handler instanceof SessionHandler && getHandler() instanceof SessionHandler)
+            setSessionHandler((SessionHandler)handler);
+        else if (handler instanceof SecurityHandler && getHandler() instanceof SecurityHandler)
+            setSecurityHandler((SecurityHandler)handler);
+        else if (handler instanceof ServletHandler && getHandler() instanceof ServletHandler)
+            setServletHandler((ServletHandler)handler);
```

Comment:
I don't think this logic is right.
Calling `setHandler()` with one of the 3 handlers will never work.

Reply:
It has been this way for several years at least.  So we are not making it special. It is already special.
I'm not sure we want to put a breaking change into the API without thought.   I found a few places in the code that did rely on this behaviour, so other users may also.

I think if we want to change setHandler, that should be a different PR with more consideration.

Reply:
@sbordet I've put the implementation of setHandler back to exactly what it has been for some time.  However, this PR removes any usage of it that depends on that behaviour.    If it is an issue, then we can make the method always warn if called on SCH.